<!-- 

-->
<launch>
    <!--env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find neuro_explorer)/config/ffd_rosconsole.conf"/-->

    <!-- launch octomap -->
    <!--include file="$(find octomap_server)/launch/octomap_mapping.launch"/-->

    <arg name="namespace" default=""/>
    <arg name="numthreads" default="8"/>
    <arg name="slam_method" default="gtmapping"/> <!-- "gmapping" -->
    <arg name="lambda" default=""/>
    <!--param name="use_sim_time" value="true"/--> <!--by kmHan-->  
    <!--param name="slam_method" value="$(arg slam_method)"/-->
    <param name="numthreads" value="$(arg numthreads)"/>
    
    <arg name="rviz_config" value ="$(find neuro_explorer)/launch/includes/rviz_config/neuro_explorer_wgx3.rviz"/>
    <rosparam command="load" file="$(find neuro_explorer)/param/neuroexploration.yaml"/>
    
    <!--include file="$(find neuro_explorer)/launch/includes/explore_bench-corridor.launch"/-->
    
    <node pkg="neuro_explorer" type="neuro_explorer_node" name="neuroexplorer" args="$(arg slam_method) $(arg numthreads) $(arg lambda)" output="screen">
        <param name="lambda" value="$(arg lambda)" />
    </node>
    <!--include file="$(find path_tracker)/launch/path_tracker.launch" output="screen"/-->
    <!--node pkg="neuro_explorer" type="neuro_explorer_node" name="neuro_explorer" output="screen">
        <param name="frame_id" type="string" value="map" />
    </node-->
    <node pkg="neuro_explorer" type="viz_helper_node" name="viz_helper" />

    <node name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(arg rviz_config)"/>    

    <!-- bag -->
    <!--node pkg="rosbag" type="record" name="rosbag_record" args="record -a -o /media/hankm/4e250950-92e8-4cb2-8ba4-b67f04a12cf6/results/neuroexplorer/bags/"/-->
    
    <!-- bag -->
    <!--node pkg="rosbag" type="record" name="rosbag_record" args="record -a -o /media/data/results/explore_bench/bag/"/-->
    
    
</launch>
